tiered-storage/jcloud/src/test/java/org/apache/bookkeeper/mledger/offload/jcloud/impl/BlockAwareSegmentInputStreamTest.java

```diff
@@ -226,7 +236,12 @@ public void testHaveEndPadding() throws Exception {
         // verify read inputStream
         // 1. read header. 128
         byte headerB[] = new byte[DataBlockHeaderImpl.getDataStartOffset()];
-        ByteStreams.readFully(inputStream, headerB);
+        if (useBufferRead) {
+            int ret = inputStream.read(headerB, 0, DataBlockHeaderImpl.getDataStartOffset());
```

Comment:
do we have a test in which the offset is not 0 ?
the risk is that is works only with 0 (and we are not handling well the offset)

Reply:
Added new test to cover this